### PR TITLE
Bugfix/make keycloak run

### PR DESCRIPTION
**Pull request by ekwanNTT**
This change makes it possible to start up and login to the local keycloak. There were also some updates to the readme to be more accurate.

### CODE REVIEW

File: settings/keycloak/keycloak-local-user.sh [hunk 1]
```diff
@@ -9,4 +9,4 @@ kcadm.sh config credentials --server http://localhost:8080/auth --realm master -
 USER_ID=`kcadm.sh create users -r mds -s username=admin -i`
 kcadm.sh update users/$USER_ID -r mds -s enabled=true
 kcadm.sh set-password -r mds --username admin --new-password admin
-kcadm.sh add-roles -r mds --uusername admin --rolename mds_application_admins --rolename mds-mine-admin
```
Comment (ekwanNTT) [inline settings/keycloak/keycloak-local-user.sh:12]: Is this role needed? It does not exist in the local realm file.

File: docker-compose.yml [hunk 2]
```diff
@@ -2,23 +2,23 @@ version: "3"
...
-  #     - DB_VENDOR=h2
-  #   volumes:
-  #     - ./monitoring/keycloak/keycloak-local-realm.json:/tmp/keycloak-local-realm.json
-  #     - ./monitoring/keycloak/keycloak-local-user.sh:/tmp/keycloak-local-user.sh
-  #   ports:
-  #     - "8080:8080"
-  #     - "8443:8443"
-  #   # logging:
-  #   #     driver: none
+  keycloak:
```
Comment (ekwanNTT) [inline docker-compose.yml:5]: Do we want to keep this uncommented now that it functions or comment it out?

**Comment by areyeslo** [comment]
Hi @ekwanNTT ,

I am checking your PR, very good work :)  the PR is related to Keycloak using OAuth2 - Openid-connect. 

I can verify the creation of admin user under keycloak-local-user.sh.  However, I am getting some issues during authentication:
```
23:47:08,185 WARN  [org.keycloak.events] (default task-27) type=LOGIN_ERROR, realmId=mds, clientId=mines-application-local, userId=null, ipAddress=172.26.0.1, **error=user_not_found**, auth_method=openid-connect, auth_type=code, redirect_uri=http://localhost:3000/home/, code
_id=356f44bc-cd75-4930-b934-33b76ac4f582, username=admin, authSessionParentId=356f44bc-cd75-4930-b934-33b76ac4f582, authSessionTabId=e33fxrJKuhE
```

As the code is in the PR, when starting localhost:3000, it is redirected to test:
image

I assumed the previous behaviour is driven by environment.js. If I replace keycloak_url: "https://test.oidc.gov.bc.ca/auth" for
keycloak_url: "http://localhost:8080/auth". In addition, I am not sure if idpHint: "idir" should be a good value. I changed this for idpHint: "admin".


After the above changes, I got the following:
image


The following is showing the Keycloak authentication and the step causing issues:
image

Can you please provide some insight how to test this change? 


Thanks

**Comment by ekwanNTT** [comment]
> Hi @ekwanNTT ,
> 
> I am checking your PR, very good work :) the PR is related to Keycloak using OAuth2 - Openid-connect.
> 
> I can verify the creation of admin user under keycloak-local-user.sh. However, I am getting some issues during authentication:
> 
> ```
> 23:47:08,185 WARN  [org.keycloak.events] (default task-27) type=LOGIN_ERROR, realmId=mds, clientId=mines-application-local, userId=null, ipAddress=172.26.0.1, **error=user_not_found**, auth_method=openid-connect, auth_type=code, redirect_uri=http://localhost:3000/home/, code
> _id=356f44bc-cd75-4930-b934-33b76ac4f582, username=admin, authSessionParentId=356f44bc-cd75-4930-b934-33b76ac4f582, authSessionTabId=e33fxrJKuhE
> ```
> 
> As the code is in the PR, when starting localhost:3000, it is redirected to test:
> image
> 
> I assumed the previous behaviour is driven by environment.js. If I replace keycloak_url: "https://test.oidc.gov.bc.ca/auth" for
> keycloak_url: "http://localhost:8080/auth". In addition, I am not sure if idpHint: "idir" should be a good value. I changed this for idpHint: "admin".
> 
> After the above changes, I got the following:
> image
> 
> The following is showing the Keycloak authentication and the step causing issues:
> image
> 
> Can you please provide some insight how to test this change?
> 
> Thanks

Hi, I believe you have to change it in more places than just the environmental file. What creds are you using to try to login to keycloak? @areyeslo

**Comment by vyasworks** [comment]
We don't use keycloak locally right now.

**Pull request closed without merging**